ts/src/index.ts

```diff
@@ -129,13 +130,34 @@ export async function start(config: Config = {}): Promise<void> {
         config);
     return;
   }
+
+  // Start the heap profiler if profiler config does not indicate heap profiling
+  // is disabled. This must be done before any asynchronous calls are made so
+  // samples from the first tick can be captured.
+  if (!config.disableHeap) {
+    heapProfiler.set(
```

Comment:
Requesting some changes:
* Remove set() method, pass parameters to enable().
* Rename enable() -> start(), disable() -> stop().
* Make stop() no op if start() wasn't called. Document it so. Remove isEnabled() method and just call stop() when needed.

This all should simplify code.